### DISCUSSION TITLE
Preset values on data object creation

### DISCUSSION THREAD

**Discussion by akno77** [pimcore/pimcore#9671] [Q&A / Support]
Hi,

I have a data object class defined with a "name" field and would like to preset the "name" field value with the "key" entered on creation. How can I achive it?

I tried to make my own model class which extends the pimcore model and implement my own "getName" method with fallback on the key, but the "getName" method is not called for the edit dialog. It does only work in the grid view...

**Comment by NiklasBr** [comment]
We have used [Event Listeners](https://pimcore.com/docs/pimcore/current/Development_Documentation/Extending_Pimcore/Event_API_and_Event_Manager.html) on [DataObjectEvents](https://github.com/pimcore/pimcore/blob/10.x/lib/Event/DataObjectEvents.php#L20-L32) previously to do that.

**Reply by akno77** [reply]
Thanks
